app/js/actions/artifacts.js

```diff
@@ -7,11 +14,78 @@ export const expectingArtifact = () => ({
     type: 'EXPECTING_ARTIFACT'
 });
 
-export const deleteArtifact = (uuid) => ({
+export const hiddenDeleteArtifact = (uuid) => ({
     type: 'DELETE_ARTIFACT',
     uuid
 });
 
-export const refreshArtifacts = () => ({
-    type: 'REFRESH_ARTIFACTS'
-});
+export const deleteArtifact = (uuid) => {
+    return (dispatch, getState) => {
+        const { artifacts: { artifacts },
+                connection: { uri, availableApis, secretKey }
+        } = getState();
+        const httpVerb = 'DELETE';
+        const timestamp = Date.now();
+        const artifact = artifacts.filter(a => a.uuid === uuid)[0];
```

Comment:
Oh, nice! There are actually a bunch of places in the code using the `filter...[0]` technique. I'll change them all.